src/main/scala/is/hail/variant/AltAlleleView.scala

```diff
@@ -0,0 +1,22 @@
+package is.hail.variant
+
+import is.hail.annotations._
+import is.hail.expr._
+import is.hail.utils._
+
+class AltAlleleView(variant: TAltAllele, source: TStruct) {
+  private val view = new StructView(variant.representation, source)
+
+  def setRegion(rv: RegionValue) {
+    view.setRegion(rv)
+  }
+
+  def setRegion(m: MemoryBuffer, offset: Long) {
+    view.setRegion(m, offset)
+  }
+
+  def hasRef(): Boolean = view.hasField("ref")
+  def getRef(): String = view.getStringField("ref")
```

Comment:
I'm skeptical of this interface if you have to do a table lookup on access.  Staging is one solution, but we're certainly going to need to access things like variants in code that doesn't end up staged.

Reply:
This was a mistake, I forgot to check in the pre-looked-up operations.